### PR TITLE
:bug: Source Plaid: min updated datetime support added for CapitalOne institutes

### PR DESCRIPTION
**Pull request by ahmed-buksh**
## What
Added support for `min_last_updated_datetime` in `Plaid Source` which is particularly required for syncing `CapitalOne` items. It indicates oldest acceptable balance as per [plaid documentation](https://plaid.com/docs/api/products/balance/#accounts-balance-get-request-options-min-last-updated-datetime).

Ticket for the issue can be found [here](https://github.com/airbytehq/airbyte/issues/10257).

Screen Shot 2022-05-25 at 15 34 16

## How
- Added field for `min_last_updated_datetime`
- Optional argument added to `AccountsBalanceRequest`

## Recommended reading order
1. `airbyte-integrations/connectors/source-plaid/source_plaid/spec.json`
2. `airbyte-integrations/connectors/source-plaid/source_plaid/source.py`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Acceptance</strong></summary>

Screen Shot 2022-05-25 at 15 39 38

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@ahmed-buksh thanks for the contribution. Please can you describe more about the problem you're trying to fix? Any links to Plaid/CapitalOne documentation describing the parameter. 
Also: why you can use the start_date parameter to this variable?

File: docs/integrations/sources/plaid.md [hunk 1]
```diff
@@ -66,6 +66,6 @@ This guide will walk through how to create the credentials you need to run this
 
 | Version | Date | Pull Request | Subject |
 | :--- | :--- | :--- | :--- |
+| 0.3.2 | 2022-05-25 | [13169](https://github.com/airbytehq/airbyte/pull/13169) | Capital One related field support added |
```
Comment (marcosmarxm) [inline docs/integrations/sources/plaid.md:69]: Please describe in Getting Started section this parameter and why it's needed.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by ahmed-buksh** [comment]
> @ahmed-buksh thanks for the contribution. Please can you describe more about the problem you're trying to fix? Any links to Plaid/CapitalOne documentation describing the parameter. Also: why you can use the start_date parameter to this variable?

@marcosmarxm I added the plaid documentation link in PR description. I guess your question is that why i didn't use `start_date` parameter to this field? In this case we don't want user to be unaware of start_date being used in that way and moreover we have both dates in different formats. For sure we can parse it to the desired format but I thought that to make it explicitly highlighted will make more sense.

This is the [ticket](https://github.com/airbytehq/airbyte/issues/10257) where the bug was reported.

**Comment by marcosmarxm** [comment]
> > @ahmed-buksh thanks for the contribution. Please can you describe more about the problem you're trying to fix? Any links to Plaid/CapitalOne documentation describing the parameter. Also: why you can use the start_date parameter to this variable?
> 
> @marcosmarxm I added the plaid documentation link in PR description. I guess your question is that why i didn't use `start_date` parameter to this field? In this case we don't want user to be unaware of start_date being used in that way and moreover we have both dates in different formats. For sure we can parse it to the desired format but I thought that to make it explicitly highlighted will make more sense.
> 
> This is the [ticket](https://github.com/airbytehq/airbyte/issues/10257) where the bug was reported.

Yes but you can reduce the complexity to users if you convert start_date to the correct format than adding a new field in specification @misteryeo can you check this DX change in Plaid connector?

File: airbyte-integrations/connectors/source-plaid/source_plaid/source.py [hunk 2]
```diff
@@ -50,7 +61,9 @@ def read_records(
         stream_slice: Mapping[str, Any] = None,
         stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
-        balance_response = self.client.accounts_balance_get(AccountsBalanceGetRequest(access_token=self.access_token))
+        options = AccountsBalanceGetRequestOptions()
+        options.min_last_updated_datetime = self.min_last_updated_datetime
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-plaid/source_plaid/source.py:65]:
```suggestion
        options.min_last_updated_datetime = datetime.format(self.start_date, "YYYY-MMTHH:MM:SSZ")
```
You can apply this formating to correct datetime the endpoints is waiting for. So users don't need to know if they CapitalOne or not, is going to work anyway.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by misteryeo** [comment]
I would agree with @marcosmarxm here @ahmed-buksh.

Our goal is to have less fields for a user to fill in...and if a new field must be created, it shouldn't be a different format if the data type is the same. 

Are you able to elaborate more on why the two fields can't just be merged into one? On the formatting, is there actually a use case where you have to set a granular time (as opposed to just the date) for the minimum value?

**Comment by marcosmarxm** [comment]
Closing due inactivity @ahmed-buksh let me know if you want to continue working on this then I can reopen it.

**Pull request closed without merging**